idn/punycode.go

```diff
@@ -267,3 +272,32 @@ func decode(b []byte) []byte {
 	}
 	return ret.Bytes()
 }
+
+// isValidRune checks if the character is valid. We will look for the
+// character property in the IDNA table. For now we aren't checking special
+// rules in case of contextual property
+func isValidRune(r rune) bool {
+	return findProperty(r) == idnaPropertyPVALID
```

Comment:
is 'idna' really important? we are already in the idn package, so dropping 'idna' makes sense?